### DISCUSSION TITLE
submit form with errors

### DISCUSSION THREAD

**Discussion by deepgolani4** [react-hook-form/react-hook-form#9091] [General] [answered]
I am trying to implement a react-hook-form but the error will be there but it shouldn't block form submission.
basically, I want to show only a warning using yup and RHF validation rules.
is there any way this is possible?
thanks for reading.

**Comment by leapful** [comment, marked as answer]
@deepgolani4 : The function "handleSubmit" returned from "useForm" hook accepts 2 callback "onValid" and "onError", you can submit your form inside "onError" callback then get the form values by using "getValues" method.

For example: 

```
const { handleSubmit, getValues } = useForm();

const onValid = () => {
   // Do something with valid form values
}

const onError = () => {
   // Continue submitting invalid form values 
   fetch('API_URL', { body: JSON.stringify(getValues()) })
}

const onSubmit = handleSubmit(onValid, onError);
```

**Reply by deepgolani4** [reply]
@leapful thank you very much
